pkg/controller/utils/component_test.go

```diff
@@ -1174,3 +1294,91 @@ func (c *fakeComponent) Objects() ([]client.Object, []client.Object) {
 func (c *fakeComponent) SupportedOSType() rmeta.OSType {
 	return c.supportedOSType
 }
+
+type mockReturn struct {
+	Method string
+	Return interface{}
+	Obj    client.Object
+}
+
+type mockClient struct {
+	Info  []mockReturn
+	Index int
+}
+
+func (mc *mockClient) Get(ctx context.Context, key client.ObjectKey, obj client.Object) error {
+	defer func() { mc.Index++ }()
+	funcName := "Get"
+	if len(mc.Info) <= mc.Index {
+		panic(fmt.Sprintf("mockClient Info doesn't have enough entries for %s %v", funcName, key))
+	}
+	if mc.Info[mc.Index].Method != funcName {
+		panic(fmt.Sprintf("mockClient current (%d) call is for %v, not %s", mc.Index, mc.Info[mc.Index].Method, funcName))
+	}
+	if mc.Info[mc.Index].Return == nil {
+		obj = mc.Info[mc.Index].Obj
+		return nil
+	}
+
+	v, ok := mc.Info[mc.Index].Return.(error)
+	if !ok {
+		panic(fmt.Sprintf("mockClient Info didn't have right type for entry %d for %s %v", mc.Index, funcName, key))
+	}
+
+	return v
+}
+
+func (mc *mockClient) List(ctx context.Context, list client.ObjectList, opts ...client.ListOption) error {
+	panic("List not implemented in mockClient")
```

Comment:
The changes in this file have unreachable return statements and unused function params.

Reply:
These are likely here to meet the interface requirements without having to fully implement the interface, i.e. he only needs `Get` and `Update` for his tests so `Get` and `Update` are the only ones implemented.

Reply:
You need either panic, or return, not both.

Reply:
That's true but the function parameters are required to meet the interface definition.

Reply:
The types of the params are, not the names.